frontend/src/GenerateHeats/GenerateHeats.jsx

```diff
@@ -78,27 +102,65 @@ const SelectAthlete = ({ eventName, eventId, onBack }) => {
     };
   }, []);
 
-  //What is needed for the itemPaginationBar
-  const handleConfirmSeedTime = () => {
-    console.log("Confirm seed time clicked!");
+  //Event Handlers
+  const handleCurrentStep = () => {
+    setAreAthletesSelected(!areAthletesSelected);
   };
-  const label = "Event " + eventName;
+
+  const handleGenerateHeats = () => {
+    console.log("Generate Heats clicked!");
+  };
+
+  const handleEditClick = (row) => {
+    setAthleteToUpdate(row);
+    setIsFormOpen(true);
+  };
+
+  const handleUpdateSeedTime = async (updatedAthlete) => {
+    setSelectedAthletes((prevAthletes) =>
+      prevAthletes.map((athlete) =>
+        athlete.id === updatedAthlete.id ? updatedAthlete : athlete
+      )
+    );
+    // Delay closing the form
+    setTimeout(() => {
+      setIsFormOpen(false);
+    }, 1500);
+  };
+
+  const actions = [
+    {
+      name: "Edit",
+      icon: <EditIcon />,
+      onClick: handleEditClick,
+      tip: "Edit Seed Time",
+    },
+  ];
+
   const extraButtons = [
     {
-      label: "Confirm seed times",
-      icon: <TimeIcon />,
-      onClick: handleConfirmSeedTime,
-      disabled: selectedAthletes.length === 0,
+      label: areAthletesSelected
+        ? "Back To Select Athletes"
+        : "Confirm seed times",
+      icon: areAthletesSelected ? <ChecklistIcon /> : <TimeIcon />,
+      onClick: handleCurrentStep,
+      disabled: !areAthletesSelected ? selectedAthletes.length === 0 : false,
     },
     {
       label: "Back to events",
       icon: <BackIcon />,
       onClick: onBack,
     },
+    {
```

Comment:
Could you please move this button to the beginning of the list?

Reply:
Yes, the Create Heats is now the first one